app/components/buttons/primary_button_component.rb

```diff
@@ -0,0 +1,10 @@
+class Buttons::PrimaryButtonComponent < ViewComponent::Base
+    def initialize(name: "", value: , type: "regular", style: "regular", color: "primary", onclick: "")
+        @name = name
+        @value = value
+        @type = type
+        @style = style
+        @color = color
+        @onclick = onclick
```

Comment:
no need to do the `@onclick` here;  the container can catch the event because it is propagated